src/uct/ib/ud/base/ud_ep.h

```diff
@@ -238,6 +238,7 @@ struct uct_ud_ep {
     ucs_wtimer_t     slow_timer;
     uint8_t          flags;
     uint8_t          path_bits;
+    ucs_queue_elem_t fail_queue;
```

Comment:
I don't think so. It may corrupt the list if failed (by timeout) but not destroyed ep gets some event asynchronously.